### PR TITLE
Remove Python 2 compatibility from numba.core.utils

### PR DESCRIPTION
**Pull request by eric-wieser**

### CODE REVIEW

**Comment by stuartarchibald** [comment]
note to self, check `LICENSE.*` for dropping six ref.

**Comment by eric-wieser** [comment]
Remove `six` and `funcsigs` from the license, good thinking

**Comment by stuartarchibald** [comment]
hmmm, segfault on the TBB build, this may be https://github.com/numba/numba/issues/5973, will take a look, it's unlikely to be changes in this PR but you never know!

**Comment by stuartarchibald** [comment]
https://github.com/numba/numba/pull/6171 is also failing and that's a changelog update, so the issue is not related to the code in here.

File: numba/core/utils.py [hunk 1]
```diff
@@ -146,7 +75,8 @@ def safe_relpath(path, start=os.curdir):
     'is': operator.is_,
     'is not': operator.is_not,
     # This one has its args reversed!
-    'in': operator.contains
+    'in': operator.contains,
+    '@': operator.matmul,
```
Comment (EPronovost) [inline numba/core/utils.py:79]: Are you planning to add support for the `@` operator on numpy arrays (probably not in this PR)?

Comment (eric-wieser) [reply numba/core/utils.py:79]: No, I just moved this from a line further down this file.

Comment (stuartarchibald) [reply numba/core/utils.py:79]: `@` operator already works on NumPy arrays IIRC.
```pycon
In [1]: from numba import njit

In [2]: import numpy as np

In [3]: a = np.ones((3, 3))

In [4]: @njit
   ...: def foo(a):
   ...:     return a@a
   ...: 

In [5]: foo(a)
Out[5]: 
array([[3., 3., 3.],
       [3., 3., 3.],
       [3., 3., 3.]])
```

**Review by gmarkall — APPROVED** [review]
Looks like this is good but just needs CI re-running. @stuartarchibald Can you re-run it please?

**Review by sklam — APPROVED** [review]
Verified the changes regarding the license file and that corresponding parts have been removed

**Pull request merged**